CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ option(NATS_BUILD_STREAMING "Build NATS Streaming" ON)
 option(NATS_BUILD_NO_PREFIX_CONNSTS "No prefix for connection status enum" OFF)
 option(NATS_BUILD_LIB_STATIC "Build static library" ON)
 option(NATS_BUILD_LIB_SHARED "Build shared library" ON)
+option(NATS_BUILD_LIB_SHARED_STATIC "Build statically-linked shared library" OFF)
```

Comment:
Looks like this is available only for CMake 3.13+. Without bumping and if you set the option to ON, you would get something like:
```
CMake Error at src/CMakeLists.txt:63 (target_link_options):
  Unknown CMake command "target_link_options".
```
But since this is OFF by default, it will not break users existing setups, if we bump requirement to 3.13 we may (but requirement to 2.8 is kind of too old anyway).

Reply:
Indeed adding target_link_option enforces cmake 3.13+. I haven't noticed this due to usage of recent cmake.
If you don't mind I would add cmake_minimum_required(VERSION 3.13).
PRIVATE here is what I use in modern cmake (verified, the keyword was not added later after 3.13) to keep -static private to the target and not to make it available for the other targets that use target_link_libraries(xxx PUBLIC -lnats)